### PR TITLE
Don't add multiple constructor declarations when using decorator @Inject() for some dependency

### PR DESCRIPTION
**Pull request by Platonn**

### CODE REVIEW

File: _pages/contributing/breaking-changes.md [hunk 1]
```diff
@@ -104,3 +104,48 @@ Aside from the above requirements, we also encourage you to do the following:
...
+```typescript
+import { PLATFORM_ID } from '@angular/core';
+/*...*/
+
+  // Include only the constructor definition, with the `?` near the new dependency:
+  constructor(
+    @Inject(PLATFORM_ID) protected platformId: any,
+    protected newService?: NewService
+  ) {}
+```
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md:151]: Now I can see I went to far with too many examples... What about dropping this?

File: _pages/contributing/breaking-changes.md [hunk 2]
```diff
@@ -104,3 +104,48 @@ Aside from the above requirements, we also encourage you to do the following:
...
+  // Do not add any constructor declarations when using @Inject to resolve a dependency
+  constructor(
+    platformId: any, // this dependency will not be resolved, nor can it be fixed with @Inject, because the Inject decorator is not supported here!
+    newService?: NewService
+  ) {}
+   constructor(
+    protected platformId: any,
+  ) {}
+
+  // Include only the constructor definition, with the `?` near the new dependency:
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md:133]:
```suggestion
```
Although my code comment here is correct, it's not related to the topic of `@Inject`. Let's drop it.

File: _pages/contributing/breaking-changes.md (hunk 1 above)
Comment (Platonn) [inline _pages/contributing/breaking-changes.md:151]:
```suggestion
```

We already explained that only constructor declaration should be used. Now I think I went too far too many code snippets. For clarity and brevity, let's drop it.

File: _pages/contributing/breaking-changes.md [hunk 3]
```diff
@@ -104,3 +104,48 @@ Aside from the above requirements, we also encourage you to do the following:
...
+
+### Using the Inject Decorator for Dependencies
+
+You should not include any constructor declarations when using `@Inject` for dependencies. Instead, you should only include the constructor definition.
+
+When you build libraries (for example, when you run `ng build --prod core`), the `ng-packagr` tool only uses the first constructor declaration to resolve injected dependencies, and the construction definition is ignored. However, the `Inject` decorator is not supported in constructor declarations, so it cannot be used to resolve dependencies there. If you include a constructor declaration with a dependency, the `ng-packagr` tool will fail to resolve the dependency and you will get an error, as follows:
```
Comment (Platonn) [inline _pages/contributing/breaking-changes.md:112]:
```suggestion
When you build libraries (for example, when you run `ng build --prod core`), the `ng-packagr` tool only uses the first constructor declaration to resolve injected dependencies, and the constructor definition is ignored. However, the `Inject` decorator is not supported in constructor declarations, so it cannot be used to resolve dependencies there. If you include a constructor declaration with a dependency, the `ng-packagr` tool will fail to resolve the dependency and you will get an error, as follows:
```
typo

File: _pages/contributing/breaking-changes.md (hunk 1 above)
Comment (gladius-mtl) [reply _pages/contributing/breaking-changes.md:151]: Discussed offline. We decided to keep it for now. Will review again when the whole page is refactored.

**Review by gladius-mtl — APPROVED** [review]

**Pull request merged**